src/common/bench_thread_switch_yield_test.c

```diff
@@ -173,6 +174,7 @@ void bench_thread_yield(void *arg)
 
 	for (i = 1; i < ITERATIONS; i++) {
 		gather_set2_stats(MAIN_PRIORITY, i);
+		bench_collect_resources();
```

Comment:
Even here, IIUIC, the ancillary thread (`bench_set2_helper`) doesn't ever get to the `bench_thread_exit()`, it's aborted before - am I missing something?

Reply:
My debugging shows that bench_thread_exit is called 1000times (or ITERATIONS times). Which effectively leads to buffer overflow. (I used different FreeRTOS_config.h than the one provided)

I have followed how other tests were organized - gather_xx_stats, bench_collect_resources. When this same template was applied on the failing test, it started to work. I have not analyzed how gather_set[12]_stats works for the yield test. 

The test framework should not assume what bench does on the inside.

Reply:
You indicated that a different configuration file was used. Can you identify the key differences?

Reply:
The issue occurs when configUSE_TIME_SLICING is defined 1 in FreeRTOSConfig.h . The bench_set2_helper() will run into bench_thread_exit() after the time slice of bench_thread_yield() is used up.

Reply:
Nice catch, I didn't notice this option, as it was not in my config file and defaults to being enabled by default. I didnt have enough time to get back to testing this.